fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneOnlineIndexingTest.java

```diff
@@ -937,5 +941,146 @@ void testRecordUpdateReducedMergeForcingAgileTimeQuota() {
+            mergeControl.setLastStep(IndexDeferredMaintenanceControl.LastStep.REBALANCE);
+            final int documentCount = mergeControl.getRepartitionDocumentCount();
+            if (documentCount <= 0) {
+                mergeControl.setRepartitionDocumentCount(16);
+                throw new FDBException("transaction_too_old", FDBError.TRANSACTION_TOO_OLD.code());
+            }
+            if (documentCount > 2) {
+                throw new FDBException("transaction_too_old", FDBError.TRANSACTION_TOO_OLD.code());
+            }
+            return AsyncUtil.DONE;
+        }
+    }
+
+    @AutoService(IndexMaintainerFactory.class)
+    public static class TerribleRebalalceIndexMaintainerFactory implements IndexMaintainerFactory {
+        @Nonnull
+        @Override
+        public Iterable<String> getIndexTypes() {
+            return Collections.singletonList("terribleRebalance");
+        }
+
+        @Nonnull
+        @Override
+        public IndexValidator getIndexValidator(Index index) {
+            return new IndexValidator(index);
+        }
+
+        @Nonnull
+        @Override
+        public IndexMaintainer getIndexMaintainer(@Nonnull IndexMaintainerState state) {
+            return new TerribleRebalalceIndexMaintainer(state);
+        }
+    }
+
+    @Test
+    void luceneOnlineIndexingTestTerribleRebalance() {
+        Index index = new Index("Simple$text_suffixes",
+                function(LuceneFunctionNames.LUCENE_TEXT, field("text")),
+                "terribleRebalance",
+                ImmutableMap.of(IndexOptions.TEXT_TOKENIZER_NAME_OPTION, AllSuffixesTextTokenizer.NAME));
+
+        boolean needMerge = populateDataSplitSegments(index, 4, 1);
+        assertTrue(needMerge);
+        try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                .setRecordStore(recordStore)
+                .setIndex(index)
+                .setMaxAttempts(1)
+                .build()) {
+            indexBuilder.mergeIndex(); // retries under the hood until document count reaches minimum
+            assertTrue(recordStore.getIndexDeferredMaintenanceControl().getRepartitionDocumentCount() <= 2);
+        }
+    }
+
+    private static class TerribleRebalalce2ndChanceIndexMaintainer extends LuceneIndexMaintainer {
+        private final IndexMaintainerState state;
+        static boolean isSecondPass = false;
+        static boolean gotSecondChance = false;
+
+        protected TerribleRebalalce2ndChanceIndexMaintainer(final IndexMaintainerState state) {
+            super(state, state.context.getExecutor());
+            this.state = state;
+        }
+
+        @Override
+        public CompletableFuture<Void> mergeIndex() {
+            final IndexDeferredMaintenanceControl mergeControl = state.store.getIndexDeferredMaintenanceControl();
+            if (isSecondPass) {
+                mergeControl.setLastStep(IndexDeferredMaintenanceControl.LastStep.REBALANCE);
+                gotSecondChance = true;
```

Comment:
I think this test would pass, if, after doing the merge, it continued to tell the indexer to skip.
Instead it should store the limit in the second chance, and assert that it is not negative.